src/backend/dx12/src/command.rs

```diff
@@ -591,25 +591,36 @@ impl CommandBuffer {
         // Use needs_bind array to determine which buffers still need to be bound
         // and bind them one continuous group at a time.
         {
-            let vbs_needs_bind = &self.vertex_buffer_views_needs_bind;
+            let vbs_remap = &self.vertex_bindings_remap;
             let vbs = &self.vertex_buffer_views;
             let mut last_end_slot = 0;
             loop {
-                match vbs_needs_bind[last_end_slot..]
+                match vbs_remap[last_end_slot..]
                     .iter()
-                    .position(|needs_bind| *needs_bind)
+                    .position(|remap| remap.is_some())
                 {
                     Some(start_offset) => {
                         let start_slot = last_end_slot + start_offset;
-                        let num_views = vbs_needs_bind[start_slot..]
+                        let num_views = vbs_remap[start_slot..]
                             .iter()
-                            .position(|needs_bind| !*needs_bind)
-                            .unwrap_or(vbs_needs_bind.len());
+                            .position(|remap| remap.is_none())
+                            .unwrap_or(vbs_remap.len());
+                        let buffers = (start_slot..start_slot+num_views).map(|i| {
+                            let remapping = vbs_remap[i].unwrap();
+                            let view = vbs[remapping.mapped_binding];
+
+                            d3d12::D3D12_VERTEX_BUFFER_VIEW {
+                                BufferLocation: view.BufferLocation + remapping.offset as u64,
+                                SizeInBytes: view.SizeInBytes - remapping.offset,
+                                StrideInBytes: remapping.stride,
```

Comment:
doesn't it cause d3d12 validation warnings, given that offset + N * stride is outside the view range?

Reply:
nop